common/persistence/elasticsearch/esVisibilityStore.go

```diff
@@ -686,9 +686,16 @@ func (v *esVisibilityStore) getSearchResult(request *p.ListWorkflowExecutionsReq
 		rangeQuery = elastic.NewRangeQuery(es.CloseTime)
 	}
 	// ElasticSearch v6 is unable to precisely compare time, have to manually add resolution 1ms to time range.
+	// Also has to use string instead of int64 to avoid data conversion issue,
+	// 9223372036854775807 to 9223372036854776000 (long overflow)
+	if request.LatestStartTime > math.MaxInt64-oneMilliSecondInNano { // prevent latestTime overflow
+		request.LatestStartTime = math.MaxInt64 - oneMilliSecondInNano
+	}
```

Comment:
additional check on the following?
1. EarliestStartTime <=LatestStartTime
2. 
```
if request. EarliestStartTime < math.MinInt64+oneMilliSecondInNano { // prevent earliestTime overflow
	request.EarliestStartTime = math.MinInt64+oneMilliSecondInNano
}
```

Reply:
updated